File: src/bot/commands.ts
import type { Context, Telegraf } from 'telegraf';

import type { AppConfig } from '../config.js';
import { postMessageToChannel } from './handlers.js';

const COMMAND_LIST = [
  '/start - Welcome and command list',
  '/help - What this bot can do',
  '/settings - Bot preferences (placeholder)',
  '/talk - Start conversation mode (placeholder)',
  '/finish - Finish session (placeholder)',
  '/dictionary - Export dictionary (placeholder)',
  '/words - Show vocabulary stats (placeholder)',
  '/posttest - Test posting to channel',
].join('\n');

export function registerCommands(bot: Telegraf<Context>, config: AppConfig): void {
  bot.start(async (ctx) => {
    await ctx.reply(
      `Welcome to your Italian Learning Assistant starter!\n\nAvailable commands:\n${COMMAND_LIST}`,
    );
  });

  bot.command('help', async (ctx) => {
    await ctx.reply(
      'This bot is the foundation of a future Italian learning assistant. ' +
        'For now, it supports basic commands and Telegram channel posting tests.',
    );
  });

  bot.command('settings', async (ctx) => {
    await ctx.reply('Settings are not implemented yet.');
  });

  bot.command('talk', async (ctx) => {
    await ctx.reply('Conversation mode will be added next.');
  });

  bot.command('finish', async (ctx) => {
    await ctx.reply('Session analysis will be added next.');
  });

  bot.command('dictionary', async (ctx) => {
    await ctx.reply('Dictionary export will be added next.');
  });

  bot.command('words', async (ctx) => {
    await ctx.reply('Vocabulary stats will be added next.');
  });

  bot.command('posttest', async (ctx) => {
    await ctx.reply('Trying to post a test message to the configured channel...');

    await postMessageToChannel(bot, config.telegramChannelId, 'Test post from bot');

    await ctx.reply('Test message posted successfully.');
  });
}

File: src/bot/commands.py
from telegram import Update
from telegram.ext import ContextTypes

from src.bot.handlers import post_message_to_channel


def _command_list() -> str:
    return "\n".join(
        [
            "/start - Welcome and command list",
            "/help - What this bot can do",
            "/settings - Settings placeholder",
            "/talk - Conversation mode placeholder",
            "/finish - Session analysis placeholder",
            "/dictionary - Dictionary export placeholder",
            "/words - Vocabulary stats placeholder",
            "/posttest - Test posting to channel",
        ]
    )


async def start_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text(
        "Welcome to your Italian Learning Assistant starter!\n\n"
        f"Available commands:\n{_command_list()}"
    )


async def help_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text(
        "This bot is the foundation of a future Italian learning assistant. "
        "For now, it supports basic commands and Telegram channel posting tests."
    )


async def settings_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text("Settings are not implemented yet")


async def talk_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text("Conversation mode will be added next")


async def finish_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text("Session analysis will be added next")


async def dictionary_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text("Dictionary export will be added next")


async def words_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text("Vocabulary stats will be added next")


async def posttest_command(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text(
        "Trying to post a test message to the configured channel..."
    )
    channel_id = context.bot_data["telegram_channel_id"]
    await post_message_to_channel(context, channel_id, "Test post from bot")
    await update.message.reply_text("Test message posted successfully.")
